czkawka: 4.1.0 -> 5.0.1

totoroot opened the pull request:
###### Description of changes
Update czkawka from version 4.1.0 to 5.0.1
Changelog can be found [here](https://github.com/qarmin/czkawka/releases/tag/5.0.0)
Czkawka GUI got ported to use GTK 4

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

yanganto commented:
There is a problem with gtk4.  
```
panicked at 'Tests failed to initialize gtk: BoolError { message: "Failed to initialize GTK", filename: "/build/czkawka-5.0.0-vendor.tar.gz/gtk4/src/rt.rs", function: "gtk4::rt", line: 145 }', /build/czkawka-5.0.0-vendor.tar.gz/gtk4/src/lib.rs:88:27
```

totoroot commented:
Result of `nixpkgs-review pr 183607` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>czkawka</li>
  </ul>
</details>

totoroot commented:
Here is the error log from nixpkgs-review:

```
error: builder for '/nix/store/icb4qiiylbvy0gqm4f0xmmkbhzvka79m-czkawka-5.0.0.drv' failed with exit code 101;
       last 10 log lines:
       > Fontconfig error: No writable cache directories
       > test help_functions::test::test_pixbuf_from_dynamic_image ... ok
       > Fontconfig error: No writable cache directories
       > thread '<unnamed>' panicked at 'Tests failed to initialize gtk: BoolError { message: "Failed to initialize GTK", filename: "/build/czkawka-5.0.0-vendor.tar.gz/gtk4/src/rt.rs", function: "gtk4::rt", line: 145 }', /build/czkawka-5.0.0-vendor.tar.gz/gtk4/src/lib.rs:88:27
       > note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
       > fatal runtime error: failed to initiate panic, error 5
       > error: test failed, to rerun pass '-p czkawka_gui --bin czkawka_gui'
       >
       > Caused by:
       >   process didn't exit successfully: `/build/source/target/x86_64-unknown-linux-gnu/release/deps/czkawka_gui-1c63697691bf750a --test-threads=12` (signal: 6, SIGABRT: process abort signal)
       For full logs, run 'nix log /nix/store/icb4qiiylbvy0gqm4f0xmmkbhzvka79m-czkawka-5.0.0.drv'.
error: 1 dependencies of derivation '/nix/store/9lzsr8ya0vad2m6fvjawlf0dbnjmni6g-review-shell.drv' failed to build

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/183607

1 package failed to build:
czkawka
```

There seems to be an issue with initializing GTK4.

applePrincess commented on pkgs/tools/misc/czkawka/default.nix:
```diff
@@ -35,7 +35,7 @@ rustPlatform.buildRustPackage rec {
-    gtk3
+    gtk4
   ];
 
```
according to qarmin/czkawka#790
you could write 

```suggestion

  checkInputs = [
    xvfb-run
  ];

  checkPhase = ''
    runHook preCheck
    xvfb-run cargo test
    runHook postCheck
  '';

```

0x4A6F replied:
Thanks, is building and running with these changes on my master-czkawka branch.

totoroot replied:
Thanks, it is building fine with these changes. I'll push the updated commit.

0x4A6F committed:
czkawka: 4.1.0 -> 5.0.1

totoroot commented:
I updated the PR's description, as the suggested changes by @applePrincess fixed the mentioned issues building the package and czkawka 5.0.0 builds fine now. PR is ready for review.

totoroot commented:
Result of `nixpkgs-review pr 183607` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>czkawka</li>
  </ul>
</details>

yanganto commented:
Hi, I just find out that the icon images are loaded correctly, such that only cli works.
```txt
RUST_BACKTRACE=1 czkawka_gui
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { domain: gdk-pixbuf-error-quark, code: 3, message: "Unrecognized image file format" }', czkawka_gui/src/help_functions.rs:709:67
stack backtrace:
   0: rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::result::unwrap_failed
   3: czkawka_gui::help_functions::set_icon_of_button
   4: czkawka_gui::gui_structs::gui_upper_notebook::GuiUpperNotebook::create_from_builder
   5: czkawka_gui::build_ui
   6: <O as gio::auto::application::ApplicationExt>::connect_command_line::command_line_trampoline
   7: _g_cclosure_marshal_INT__OBJECT
   8: g_closure_invoke
   9: signal_emit_unlocked_R
  10: g_signal_emit_valist
  11: g_signal_emit
  12: g_application_call_command_line
  13: g_application_real_local_command_line
  14: g_application_run
  15: <O as gio::application::ApplicationExtManual>::run_with_args
  16: czkawka_gui::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

applePrincess commented:
Mine one works too.
![screenshot](https://user-images.githubusercontent.com/17154507/182868252-e12b5165-33d9-4994-b4f9-73c325221869.png)

```
[nix-shell:~/.cache/nixpkgs-review/pr-183607-2]$ ./results/czkawka/bin/czkawka_gui

(czkawka_gui:138387): Gdk-WARNING **: 14:09:35.160: Compositor doesn't support moving popups, relying on remapping

(czkawka_gui:138387): GLib-GIO-ERROR **: 14:09:35.586: No GSettings schemas are installed on the system
Trace/breakpoint trap (core dumped)
```

0x4A6F commented:
Could you test [these changes](https://github.com/NixOS/nixpkgs/compare/master...0x4A6F:master-czkawka)?

applePrincess commented:
![screenshot](https://user-images.githubusercontent.com/17154507/182975647-a2d2b973-ebfe-4259-9e5e-806f06c10c67.png)

@0x4A6F looks good for me.  Unlike above it does not show any error or warning messages.

totoroot commented:
@0x4A6F Your changes look good so I rebased on your branch and squashed the commits.
I'll update the description, as the update is now from version 4.1.0 to 5.0.1

totoroot commented:
Result of `nixpkgs-review pr 183607` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>czkawka</li>
  </ul>
</details>
